tests/color_config.rs

```diff
@@ -0,0 +1,97 @@
+//! Tests the --color flag and CARGO_TERM_COLOR env var, and their interactions
+//!
+//! The --color flag should take precedence over the environment variable
+use assert_cmd::Command;
+use predicates::boolean::PredicateBooleanExt;
+
+/// helper function to run `check-release` on a given directory in `test_crates/` with the given color arg/env var
+/// - `color_arg` is the argument to `--color=<choice>`, if set
+/// - `color_env_var` sets the `CARGO_TERM_COLOR=choice`, if set
+///
+/// Panics with a failed assertion if there are colors (i.e., ANSI color codes) in the output and `assert_colors` is false, or vice versa
+fn run_on_crate_diff(
+    test_crate_name: &'static str,
+    color_arg: Option<&'static str>,
+    color_env_var: Option<&'static str>,
+    assert_colors: bool,
+) {
+    let mut cmd = Command::cargo_bin("cargo-semver-checks").unwrap();
+
+    cmd.current_dir(format!("test_crates/{}", test_crate_name))
+        .args([
+            "semver-checks",
+            "check-release",
+            "--manifest-path=new/Cargo.toml",
+            "--baseline-root=old/",
+        ]);
+
+    if let Some(color_arg) = color_arg {
+        cmd.arg("--color").arg(color_arg);
+    }
+
+    if let Some(color_env) = color_env_var {
+        cmd.env("CARGO_TERM_COLOR", color_env);
+    }
```

Comment:
GitHub Actions might set `CARGO_TERM_COLOR` on the process that runs `cargo test`, and in this case this command will probably inherit it. We had a flaky test for this reason in the past.

Let's explicitly un-set the env var if `color_env_var` is None.
```suggestion
    if let Some(color_env) = color_env_var {
        cmd.env("CARGO_TERM_COLOR", color_env);
    } else {
        cmd.env_remove("CARGO_TERM_COLOR");
    }
```